include/wx/private/webviewutils.h

```diff
@@ -0,0 +1,45 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/private/webviewutils.h
+// Purpose:     Common code for wxWebView
+// Author:      Jose Lorenzo
+// Created:     2017-08-12
+// Copyright:   (c) 2017 Jose Lorenzo <josee.loren@gmail.com>
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_PRIVATE_WEBVIEWUTILS_H_
+#define _WX_PRIVATE_WEBVIEWUTILS_H_
+
+#include "wx/regex.h"
+
+// ----------------------------------------------------------------------------
+// Common wxWebView code
+// ----------------------------------------------------------------------------
+
+class wxWebViewUtils
+{
+public:
+
+    // This method is used to add a double quote level into a JavasSript code
+    // in order to get it working when eval is called programmatically.
+    static wxString WrapJavaScript
+        (const wxString& javascript, int* runScriptCount, wxString* counter)
```

Comment:
This function is really confusing and difficult to use correctly.

One obvious improvement would be to return both the script to execute and the variable to check, instead of using `counter` output parameter which then needs to be appended to `__wx$` in the caller, thus leaking the details of this function implementation.

In fact, it would probably be better to have a class, except not called `wxWebViewUtils` but something like `wxJSScriptWrapper`, which you'd use like this:

    wxString result;
    wxJSScriptWrapper wrapJS(script);
    RunScriptInternal(wrapJS.GetWrapperCode(), &result);
    if ( result ok )
        RunScriptInternal(wrapJS.GetOutputCode(), &result);

This would be easier to use and much more straightforward to implement while requiring relatively few changes.